### DISCUSSION TITLE
能添加kmod-xdp-sockets-diag这个依赖么，dae安装时提示缺少这个依赖

### DISCUSSION THREAD

**Discussion by nibuhuicai** [coolsnowwolf/lede#11799] [Bugs / 问题反馈]
能添加kmod-xdp-sockets-diag这个依赖么，dae安装时提示缺少这个依赖

**Comment by sdttttt** [comment]
这个其实只要编译的时候加一下就行了。

```bash
echo '

define KernelPackage/xdp-sockets-diag
  SUBMENU:=$(NETWORK_SUPPORT_MENU)
  TITLE:=PF_XDP sockets monitoring interface support for ss utility
  KCONFIG:= \
	CONFIG_XDP_SOCKETS=y \
	CONFIG_XDP_SOCKETS_DIAG
  FILES:=$(LINUX_DIR)/net/xdp/xsk_diag.ko
  AUTOLOAD:=$(call AutoLoad,31,xsk_diag)
endef

define KernelPackage/xdp-sockets-diag/description
 Support for PF_XDP sockets monitoring interface used by the ss tool
endef

$(eval $(call KernelPackage,xdp-sockets-diag))
' >> package/kernel/linux/modules/netsupport.mk


echo '

CONFIG_PACKAGE_kmod-xdp-sockets-diag=y
' >> .config

```

**Reply by twedu** [reply]
> 这个其实只要编译的时候加一下就行了。
> 
> ```shell
> echo '
> 
> define KernelPackage/xdp-sockets-diag
>   SUBMENU:=$(NETWORK_SUPPORT_MENU)
>   TITLE:=PF_XDP sockets monitoring interface support for ss utility
>   KCONFIG:= \
> 	CONFIG_XDP_SOCKETS=y \
> 	CONFIG_XDP_SOCKETS_DIAG
>   FILES:=$(LINUX_DIR)/net/xdp/xsk_diag.ko
>   AUTOLOAD:=$(call AutoLoad,31,xsk_diag)
> endef
> 
> define KernelPackage/xdp-sockets-diag/description
>  Support for PF_XDP sockets monitoring interface used by the ss tool
> endef
> 
> $(eval $(call KernelPackage,xdp-sockets-diag))
> ' >> package/kernel/linux/modules/netsupport.mk
> 
> 
> echo '
> 
> CONFIG_PACKAGE_kmod-xdp-sockets-diag=y
> ' >> .config
> ```

官方openwrt就没法子用了吧？

**Reply by sdttttt** [reply]
官方没用过诶，不清楚
